File: 🟡 Medium/2938 - Separate Black and White Balls/_2938SeparateBlackandWhiteBalls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        res = 0
        whitePos = 0

        for zero in range(0, len(s)):
            if s[zero] == "0":
                res += zero - whitePos
                whitePos += 1

            zero -= 1

        return res

File: 🟡 Medium/2938 - Separate Black and White Balls/_2938SeparateBlackandWhiteBalls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long res = 0;
        int zero = 0;
        int whitePos = 0;

        while (zero < s.length()) {
            if (s[zero] == '0') {
                res += (zero - whitePos);
                whitePos++;
            }

            zero++;
        }

        return res;
    }
};